### PR TITLE
Use Dataset in input pipeline + other QoL changes

### PR DESCRIPTION
**Pull request by V-Xing**
Following our discussion in the Google group, this is my version of the bls2017 example. I mainly tweaked the input pipeline to use a clean Dataset+Re-initializable Iterator structure and added Tensorboard visualization.
Any feedback is welcome!
The main changes are documented in the commit messages, and are copied below. 

- The Dataset structure is used to avoid memory overload when
loading data in memory.
- Training evolution can be visualized using Tensorboard.
The loss function and its rate term are plotted for the training
and validation set.
- Runtime profiling of the compression step is implemented. A .json
file is created and can be opened using chrome://tracing.
- Data augmentation is implemented in case overfitting is observed.

### CODE REVIEW

**Comment by V-Xing** [comment]
I signed it!

**Comment by V-Xing** [comment]
Closed because of messed up commit history
